cookies/views/giles.py

```diff
@@ -328,4 +331,27 @@ def log(request):
 @auth.authorization_required(ResourceAuthorization.VIEW, _get_upload_by_id)
 def log_item(request, upload_id):
     upload = get_object_or_404(GilesUpload, pk=upload_id)
-    return render(request, 'giles_log_item.html', {'upload': upload})
+    if request.method == 'POST':
+        if auth.check_authorization(ResourceAuthorization.EDIT,
+                                    request.user,
+                                    upload.resource):
+            try:
+                giles.process_upload(upload.upload_id, upload.created_by.username)
```

Comment:
will that work if I am not the user who originally uploaded a file? Looking at 'process_upload' it seems to check if the created_by user is the same as the currently logged in user.

Reply:
also, will that created additional resources if there are already some stored for the upload?  checking the status should also be used for updating an existing upload (e.g. it might be that some components of Giles are slower in processing files and when giles considers an upload done, those might not have finished yet (additional components are not part of the overall workflow but kind of indepenedent)).

Reply:
> will that work if I am not the user who originally uploaded a file? Looking at 'process_upload' it seems to check if the created_by user is the same as the currently logged in user.

Yes. The assumption is if a user has edit privileges on a resource they can request a manual Giles status check on behalf of the user who created Giles upload.

Reply:
> also, will that created additional resources if there are already some stored for the upload?
It won't. It will see that the status is Done and return.

> checking the status should also be used for updating an existing upload (e.g. it might be that some components of Giles are slower in processing files and when giles considers an upload done, those might not have finished yet (additional components are not part of the overall workflow but kind of indepenedent)).

Got it. I will update the recheck logic.